### DISCUSSION TITLE
Could you please make an Android version...

### DISCUSSION THREAD

**Discussion by ghost** [paveloom-f/lbry-desktop#15] [General]
that removes the blacklist?

**Comment by paveloom** [comment]
Maybe when I have an Android phone.

**Comment by paveloom** [comment]
Heads up! The Android fork is [here](https://github.com/paveloom-f/lbry-android). Some testing would be appreciated, though.

**Reply by ghost** [reply]
Videos still aren't displayed on blacklisted channels but blacklisted videos do play. That's better than nothing. Thank you very much.

**Reply by ghost** [reply]
I can use https://madiator.com/ to copy links for blacklisted videos to play on the modded android app or I can copy links from the desktop version you made. What matters is that the blacklisted videos can now be played.

**Reply by paveloom** [reply]
Would you mind testing again, @ismail6172? I've just updated the binaries with a different patch. Blocked content should now be displayed on the channels.

**Reply by ghost** [reply]
Blacklisted videos are now displayed on channels. Thank you very much. I appreciate it greatly.

**Reply by ghost** [reply]
> Would you mind testing again, @ismail6172? I've just updated the binaries with a different patch. Blocked content should now be displayed on the channels.

There's another way claims can be blacklisted that I noticed recently with a channel I would occasionally watch. They can make it look like the claim was abandoned. Your desktop fork bypasses that but the Android fork doesn't. You think you can fix that? Would appreciate it.

It seems like there are two blacklists. One for copyright and one for everything else. 

lbry://@WebmPoster#e0e8aa15bb52d08c32472b794854203feaafdaa2

**Reply by ghost** [reply]
> Would you mind testing again, @ismail6172? I've just updated the binaries with a different patch. Blocked content should now be displayed on the channels.

I think you only removed the blacklist for copyrighted contented on Android. That is my point.

**Reply by paveloom** [reply]
Hello, @ismail6172. Thanks for the report!

There are indeed two lists:
- [api.lbry.io/file/list_blocked](https://api.lbry.io/file/list_blocked)
- [api.lbry.io/file/list_filtered](https://api.lbry.io/file/list_filtered)

Both should by bypassed by my patch in the Android version:

Here's the patch:

https://github.com/paveloom-f/lbry-android/blob/29b66c3d42e7c4fc76fa6847f8d5d2a6d9687955/.github/scripts/build_a_release.bash#L18-L19

Here's the diff:

```diff
diff --git a/app/src/main/java/io/lbry/browser/MainActivity.java b/app/src/main/java/io/lbry/browser/MainActivity.java
index 5bee4b9..835faf2 100644
--- a/app/src/main/java/io/lbry/browser/MainActivity.java
+++ b/app/src/main/java/io/lbry/browser/MainActivity.java
@@ -2935,12 +2935,12 @@ public class MainActivity extends AppCompatActivity implements SdkStatusListener
 
                     JSONObject blockedObject = (JSONObject) Lbryio.parseResponse(Lbryio.call("file", "list_blocked", context));
                     JSONArray blockedArray = blockedObject.getJSONArray("outpoints");
-                    Lbryio.populateOutpointList(Lbryio.blockedOutpoints, blockedArray);
+                    // Lbryio.populateOutpointList(Lbryio.blockedOutpoints, blockedArray);
                     startupStages.set(STARTUP_STAGE_BLOCK_LIST_LOADED - 1, new StartupStage(STARTUP_STAGE_BLOCK_LIST_LOADED, true));
 
                     JSONObject filteredObject = (JSONObject) Lbryio.parseResponse(Lbryio.call("file", "list_filtered", context));
                     JSONArray filteredArray = filteredObject.getJSONArray("outpoints");
-                    Lbryio.populateOutpointList(Lbryio.filteredOutpoints, filteredArray);
+                    // Lbryio.populateOutpointList(Lbryio.filteredOutpoints, filteredArray);
                     startupStages.set(STARTUP_STAGE_FILTER_LIST_LOADED - 1, new StartupStage(STARTUP_STAGE_FILTER_LIST_LOADED, true));
                 } catch (Exception ex) {
                     // nope

```

You can check if either of the lists is ignored by doing the following:
1) Take a hash ID from the list, removing the `:0` part in the end
2) [Query](https://chainquery.lbry.com/api/sql?query=SELECT%20*%20FROM%20claim%20WHERE%20transaction_hash_update%20=%20%271441007a20747b1b9f92dfac620089308268be23afe99b35d8c410dfd3aaeeb6%27) the blockchain to get metadata (change the hash in the SQL query)
3) Check the name of the claim on [Odysee.com](https://odysee.com) to see if it's actually blocked / filtered
4) Check the name of the claim in either of my forks to see if it's accessible

Here's an example from the `blocked` list:
[@GIRLSMUSIC](https://odysee.com/@GIRLSMUSIC:3) (1441007a20747b1b9f92dfac620089308268be23afe99b35d8c410dfd3aaeeb6)

You can check the `filtered` list yourself (lots of NSFW content, so I won't post it here).

I can confirm that claims from the `filtered` list are not accessible with the Android fork, despite the patch above. That's weird, because the snippet above is the only place where the `list_filtered` endpoint is queried. 

What's also weird is that I don't see the transaction hash ID of the [channel](https://chainquery.lbry.com/api/sql?query=SELECT%20*%20FROM%20claim%20WHERE%20name%20=%20%27@WebmPoster%27) you mentioned in either of the lists.

I will investigate it further.

**Reply by ghost** [reply]
> Hello, @ismail6172. Thanks for the report!
> 
> There are indeed two lists:
> - [api.lbry.io/file/list_blocked](https://api.lbry.io/file/list_blocked)
> - [api.lbry.io/file/list_filtered](https://api.lbry.io/file/list_filtered)
> 
> Both should by bypassed by my patch in the Android version:
> 
> Here's the patch:
> 
> https://github.com/paveloom-f/lbry-android/blob/29b66c3d42e7c4fc76fa6847f8d5d2a6d9687955/.github/scripts/build_a_release.bash#L18-L19
> 
> Here's the diff:
> 
> ```diff
> diff --git a/app/src/main/java/io/lbry/browser/MainActivity.java b/app/src/main/java/io/lbry/browser/MainActivity.java
> index 5bee4b9..835faf2 100644
> --- a/app/src/main/java/io/lbry/browser/MainActivity.java
> +++ b/app/src/main/java/io/lbry/browser/MainActivity.java
> @@ -2935,12 +2935,12 @@ public class MainActivity extends AppCompatActivity implements SdkStatusListener
>  
>                      JSONObject blockedObject = (JSONObject) Lbryio.parseResponse(Lbryio.call("file", "list_blocked", context));
>                      JSONArray blockedArray = blockedObject.getJSONArray("outpoints");
> -                    Lbryio.populateOutpointList(Lbryio.blockedOutpoints, blockedArray);
> +                    // Lbryio.populateOutpointList(Lbryio.blockedOutpoints, blockedArray);
>                      startupStages.set(STARTUP_STAGE_BLOCK_LIST_LOADED - 1, new StartupStage(STARTUP_STAGE_BLOCK_LIST_LOADED, true));
>  
>                      JSONObject filteredObject = (JSONObject) Lbryio.parseResponse(Lbryio.call("file", "list_filtered", context));
>                      JSONArray filteredArray = filteredObject.getJSONArray("outpoints");
> -                    Lbryio.populateOutpointList(Lbryio.filteredOutpoints, filteredArray);
> +                    // Lbryio.populateOutpointList(Lbryio.filteredOutpoints, filteredArray);
>                      startupStages.set(STARTUP_STAGE_FILTER_LIST_LOADED - 1, new StartupStage(STARTUP_STAGE_FILTER_LIST_LOADED, true));
>                  } catch (Exception ex) {
>                      // nope
> 
> ```
> 
> You can check if either of the lists is ignored by doing the following:
> 1) Take a hash ID from the list, removing the `:0` part in the end
> 2) [Query](https://chainquery.lbry.com/api/sql?query=SELECT%20*%20FROM%20claim%20WHERE%20transaction_hash_update%20=%20%271441007a20747b1b9f92dfac620089308268be23afe99b35d8c410dfd3aaeeb6%27) the blockchain to get metadata (change the hash in the SQL query)
> 3) Check the name of the claim on [Odysee.com](https://odysee.com) to see if it's actually blocked / filtered
> 4) Check the name of the claim in either of my forks to see if it's accessible
> 
> Here's an example from the `blocked` list:
> [@GIRLSMUSIC](https://odysee.com/@GIRLSMUSIC:3) (1441007a20747b1b9f92dfac620089308268be23afe99b35d8c410dfd3aaeeb6)
> 
> You can check the `filtered` list yourself (lots of NSFW content, so I won't post it here).
> 
> I can confirm that claims from the `filtered` list are not accessible with the Android fork, despite the patch above. That's weird, because the snippet above is the only place where the `list_filtered` endpoint is queried. 
> 
> What's also weird is that I don't see the transaction hash ID of the [channel](https://chainquery.lbry.com/api/sql?query=SELECT%20*%20FROM%20claim%20WHERE%20name%20=%20%27@WebmPoster%27) you mentioned in either of the lists.
> 
> I will investigate it further.

I only tested the first filter which I guess is for copyrighted content a year ago because I didn't know there was a second one. My apologies.

**Reply by ghost** [reply]
> > Hello, @ismail6172. Thanks for the report!
> > 
> > There are indeed two lists:
> > - [api.lbry.io/file/list_blocked](https://api.lbry.io/file/list_blocked)
> > - [api.lbry.io/file/list_filtered](https://api.lbry.io/file/list_filtered)
> > 
> > Both should by bypassed by my patch in the Android version:
> > 
> > Here's the patch:
> > 
> > https://github.com/paveloom-f/lbry-android/blob/29b66c3d42e7c4fc76fa6847f8d5d2a6d9687955/.github/scripts/build_a_release.bash#L18-L19
> > 
> > Here's the diff:
> > 
> > ```diff
> > diff --git a/app/src/main/java/io/lbry/browser/MainActivity.java b/app/src/main/java/io/lbry/browser/MainActivity.java
> > index 5bee4b9..835faf2 100644
> > --- a/app/src/main/java/io/lbry/browser/MainActivity.java
> > +++ b/app/src/main/java/io/lbry/browser/MainActivity.java
> > @@ -2935,12 +2935,12 @@ public class MainActivity extends AppCompatActivity implements SdkStatusListener
> >  
> >                      JSONObject blockedObject = (JSONObject) Lbryio.parseResponse(Lbryio.call("file", "list_blocked", context));
> >                      JSONArray blockedArray = blockedObject.getJSONArray("outpoints");
> > -                    Lbryio.populateOutpointList(Lbryio.blockedOutpoints, blockedArray);
> > +                    // Lbryio.populateOutpointList(Lbryio.blockedOutpoints, blockedArray);
> >                      startupStages.set(STARTUP_STAGE_BLOCK_LIST_LOADED - 1, new StartupStage(STARTUP_STAGE_BLOCK_LIST_LOADED, true));
> >  
> >                      JSONObject filteredObject = (JSONObject) Lbryio.parseResponse(Lbryio.call("file", "list_filtered", context));
> >                      JSONArray filteredArray = filteredObject.getJSONArray("outpoints");
> > -                    Lbryio.populateOutpointList(Lbryio.filteredOutpoints, filteredArray);
> > +                    // Lbryio.populateOutpointList(Lbryio.filteredOutpoints, filteredArray);
> >                      startupStages.set(STARTUP_STAGE_FILTER_LIST_LOADED - 1, new StartupStage(STARTUP_STAGE_FILTER_LIST_LOADED, true));
> >                  } catch (Exception ex) {
> >                      // nope
> > 
> > ```
> > 
> > You can check if either of the lists is ignored by doing the following:
> > 1) Take a hash ID from the list, removing the `:0` part in the end
> > 2) [Query](https://chainquery.lbry.com/api/sql?query=SELECT%20*%20FROM%20claim%20WHERE%20transaction_hash_update%20=%20%271441007a20747b1b9f92dfac620089308268be23afe99b35d8c410dfd3aaeeb6%27) the blockchain to get metadata (change the hash in the SQL query)
> > 3) Check the name of the claim on [Odysee.com](https://odysee.com) to see if it's actually blocked / filtered
> > 4) Check the name of the claim in either of my forks to see if it's accessible
> > 
> > Here's an example from the `blocked` list:
> > [@GIRLSMUSIC](https://odysee.com/@GIRLSMUSIC:3) (1441007a20747b1b9f92dfac620089308268be23afe99b35d8c410dfd3aaeeb6)
> > 
> > You can check the `filtered` list yourself (lots of NSFW content, so I won't post it here).
> > 
> > I can confirm that claims from the `filtered` list are not accessible with the Android fork, despite the patch above. That's weird, because the snippet above is the only place where the `list_filtered` endpoint is queried. 
> > 
> > What's also weird is that I don't see the transaction hash ID of the [channel](https://chainquery.lbry.com/api/sql?query=SELECT%20*%20FROM%20claim%20WHERE%20name%20=%20%27@WebmPoster%27) you mentioned in either of the lists.
> > 
> > I will investigate it further.
> 
> I only tested the first filter which I guess is for copyrighted content a year ago because I didn't know there was a second one. My apologies.

I guess with the second filter they can make it look like the channel never existed or abandoned like a shadow ban.

**Reply by ghost** [reply]
> Hello, @ismail6172. Thanks for the report!
> 
> There are indeed two lists:
> - [api.lbry.io/file/list_blocked](https://api.lbry.io/file/list_blocked)
> - [api.lbry.io/file/list_filtered](https://api.lbry.io/file/list_filtered)
> 
> Both should by bypassed by my patch in the Android version:
> 
> Here's the patch:
> 
> https://github.com/paveloom-f/lbry-android/blob/29b66c3d42e7c4fc76fa6847f8d5d2a6d9687955/.github/scripts/build_a_release.bash#L18-L19
> 
> Here's the diff:
> 
> ```diff
> diff --git a/app/src/main/java/io/lbry/browser/MainActivity.java b/app/src/main/java/io/lbry/browser/MainActivity.java
> index 5bee4b9..835faf2 100644
> --- a/app/src/main/java/io/lbry/browser/MainActivity.java
> +++ b/app/src/main/java/io/lbry/browser/MainActivity.java
> @@ -2935,12 +2935,12 @@ public class MainActivity extends AppCompatActivity implements SdkStatusListener
>  
>                      JSONObject blockedObject = (JSONObject) Lbryio.parseResponse(Lbryio.call("file", "list_blocked", context));
>                      JSONArray blockedArray = blockedObject.getJSONArray("outpoints");
> -                    Lbryio.populateOutpointList(Lbryio.blockedOutpoints, blockedArray);
> +                    // Lbryio.populateOutpointList(Lbryio.blockedOutpoints, blockedArray);
>                      startupStages.set(STARTUP_STAGE_BLOCK_LIST_LOADED - 1, new StartupStage(STARTUP_STAGE_BLOCK_LIST_LOADED, true));
>  
>                      JSONObject filteredObject = (JSONObject) Lbryio.parseResponse(Lbryio.call("file", "list_filtered", context));
>                      JSONArray filteredArray = filteredObject.getJSONArray("outpoints");
> -                    Lbryio.populateOutpointList(Lbryio.filteredOutpoints, filteredArray);
> +                    // Lbryio.populateOutpointList(Lbryio.filteredOutpoints, filteredArray);
>                      startupStages.set(STARTUP_STAGE_FILTER_LIST_LOADED - 1, new StartupStage(STARTUP_STAGE_FILTER_LIST_LOADED, true));
>                  } catch (Exception ex) {
>                      // nope
> 
> ```
> 
> You can check if either of the lists is ignored by doing the following:
> 1) Take a hash ID from the list, removing the `:0` part in the end
> 2) [Query](https://chainquery.lbry.com/api/sql?query=SELECT%20*%20FROM%20claim%20WHERE%20transaction_hash_update%20=%20%271441007a20747b1b9f92dfac620089308268be23afe99b35d8c410dfd3aaeeb6%27) the blockchain to get metadata (change the hash in the SQL query)
> 3) Check the name of the claim on [Odysee.com](https://odysee.com) to see if it's actually blocked / filtered
> 4) Check the name of the claim in either of my forks to see if it's accessible
> 
> Here's an example from the `blocked` list:
> [@GIRLSMUSIC](https://odysee.com/@GIRLSMUSIC:3) (1441007a20747b1b9f92dfac620089308268be23afe99b35d8c410dfd3aaeeb6)
> 
> You can check the `filtered` list yourself (lots of NSFW content, so I won't post it here).
> 
> I can confirm that claims from the `filtered` list are not accessible with the Android fork, despite the patch above. That's weird, because the snippet above is the only place where the `list_filtered` endpoint is queried. 
> 
> What's also weird is that I don't see the transaction hash ID of the [channel](https://chainquery.lbry.com/api/sql?query=SELECT%20*%20FROM%20claim%20WHERE%20name%20=%20%27@WebmPoster%27) you mentioned in either of the lists.
> 
> I will investigate it further.

I forgot to mention that the videos/channels do show up on search results but when you click on them it does it says nothing found as if the claim was abandoned or never existed. I hope that helps.

**Reply by paveloom** [reply]
Correction: I can see the filtered content from the `filtered` list. Not sure why I couldn't before. So, both lists are bypassed.

Looking at the code, the `There's nothing at this location` message is only rendered in these two components:

`FileViewFragment`:
https://github.com/lbryio/lbry-android/blob/master/app/src/main/java/io/lbry/browser/ui/findcontent/FileViewFragment.java#L941-L988 

`ChannelFragment`:
https://github.com/lbryio/lbry-android/blob/master/app/src/main/java/io/lbry/browser/ui/channel/ChannelFragment.java#L485-L510

Both will render this text after failing to resolve the claim using https://api.lbry.tv/api/v1/proxy. Perhaps, this is where censoring (or a serch error) happens.

Do you have other examples of channels or files you don't see in the Android app, besides @WebmPoster?

**Reply by ghost** [reply]
> Correction: I can see the filtered content from the `filtered` list. Not sure why I couldn't before. So, both lists are bypassed.
> 
> Looking at the code, the `There's nothing at this location` message is only rendered in these two components:
> 
> `FileViewFragment`:
> https://github.com/lbryio/lbry-android/blob/master/app/src/main/java/io/lbry/browser/ui/findcontent/FileViewFragment.java#L941-L988 
> 
> `ChannelFragment`:
> https://github.com/lbryio/lbry-android/blob/master/app/src/main/java/io/lbry/browser/ui/channel/ChannelFragment.java#L485-L510
> 
> Both will render this text after failing to resolve the claim using https://api.lbry.tv/api/v1/proxy. Perhaps, this is where censoring (or a serch error) happens.
> 
> Do you have other examples of channels or files you don't see in the Android app, besides @WebmPoster?

I will try to find more. I found one more so far.

lbry://@WeWuzKangz

**Reply by ghost** [reply]
> Correction: I can see the filtered content from the `filtered` list. Not sure why I couldn't before. So, both lists are bypassed.
> 
> Looking at the code, the `There's nothing at this location` message is only rendered in these two components:
> 
> `FileViewFragment`:
> https://github.com/lbryio/lbry-android/blob/master/app/src/main/java/io/lbry/browser/ui/findcontent/FileViewFragment.java#L941-L988 
> 
> `ChannelFragment`:
> https://github.com/lbryio/lbry-android/blob/master/app/src/main/java/io/lbry/browser/ui/channel/ChannelFragment.java#L485-L510
> 
> Both will render this text after failing to resolve the claim using https://api.lbry.tv/api/v1/proxy. Perhaps, this is where censoring (or a serch error) happens.
> 
> Do you have other examples of channels or files you don't see in the Android app, besides @WebmPoster?

You think you could release an Android fork with video optimization disabled? It's just transcoding but rebranded. You can't disable it when uploading videos. Not really needed as I can upload on Desktop but would be more convenient so I would appreciate it.

**Reply by paveloom** [reply]
Aight, I can confirm that this channel is also inaccessible in the Android fork.

These two channels (lbry://@WebmPoster and lbry://@WeWuzKangz) aren't in the public [`list_filtered`](https://api.lbry.io/file/list_filtered) and [`list_blocked`](https://api.lbry.io/file/list_blocked) lists, so the origin of blocking is unknown to me. At the moment, though, I do believe that these claims aren't being resolved on the server-side (the app queries https://api.lbry.tv/api/v1/proxy, as described above, which is an older version of the API). Since it happens on the server, I can't change it.

Disabling the optimization is out of scope of that repository. Some trickery with [paths](https://github.com/lbryio/lbry-android/blob/master/app/src/main/java/io/lbry/browser/ui/publish/PublishFormFragment.java#L1397-L1403) could be applied by someone interested.
